python/lsst/cell_coadds/_stitched_aperture_correction.py

```diff
@@ -46,19 +51,34 @@ def __init__(self, ugrid: UniformGrid, gc: Mapping[Index2D, float]):
         self.ugrid = ugrid
         self.gc = gc
 
-    def evaluate(self, point: geom.Point2D | geom.Point2I) -> float:
+    def evaluate(
+        self, x: geom.Point2D | geom.Point2I | Iterable[float], y: Iterable[float] | None = None
+    ) -> float | np.ndarray:
         """Evaluate the BoundedField at a given point on the image.
 
         Parameters
         ----------
-        point : `~lsst.geom.Point2D` or `~lsst.geom.Point2I`
-            The point at which to evaluate the BoundedField.
+        x : `~lsst.geom.Point2D` or `~lsst.geom.Point2I`, or `Iterable[float]`
+            The point at which to evaluate the BoundedField, or an iterable of
+            x-coordinates.
+        y : `Iterable[float]`, optional
+            The y-coordinates of the points at which to evaluate the aperture
+            correction.
+            If `None`, `x` is assumed to be a single point.
 
         Returns
         -------
-        value: `float`
+        value: `float`, or `numpy.ndarray`
             The value of the BoundedField at the specified point.
         """
-        eval_point = geom.Point2I(point)
-        idx = self.ugrid.index(eval_point)
-        return self.gc[idx]
+        if y is None:
+            eval_point = geom.Point2I(x)
+            idx = self.ugrid.index(eval_point)
+            try:
+                return self.gc[idx]
+            except KeyError:
+                logger.info("No aperture correction found for %s", idx)
+                return 1.0
+
+        else:
+            return np.array([self.evaluate(geom.Point2I(xx, yy)) for xx, yy in zip(x, y, strict=True)])
```

Comment:
We could either define two different functions if we prefer to not have a method calling itself. Or if we want to drop the support for taking a `Point2D` object, that'd be fine as well.

Reply:
I think what you've got here is fine.  Might be nice to declare `@typing.overloads` for it, but given that the coding calling it isn't type-checked, I'm not sure it's worth much.  It would let MyPy verify that the composite signature is consistent with the two overloads, though.

Reply:
I don't think it recognizes `geom.Point2I` and `geom.Point2D` types well, so I had to force it to ignore types where the method is actually implemented. All GitHub actions pass now.